src/workloads/sharding/MongosMerging.yml

```diff
@@ -0,0 +1,151 @@
+SchemaVersion: 2018-07-01
+Owner: "@mongodb/product-perf"
+Description: |
+  This workload runs different types of aggregations where the query results
+  will be merged on a mongos node.
+GlobalDefaults:
+  MaxPhases: &MaxPhases 10
+  NumDocs: &NumDocs 5000000
+  Database: &Database test
+
+  ArraySize: &ArraySize 10
+
+  SampleSizeForPseudoRandom: &SampleSizeForPseudoRandom 49999
+  SampleSizeForRandomSort: &SampleSizeForRandomSort 100000
+Clients:
+  Default:
+    QueryOptions:
+      maxPoolSize: 20000
+
+Keywords:
+- Aggregations
+- Mongos
+- Sample
+- Unwind
+- Sort
+
+ActorTemplates:
+- TemplateName: Aggregation
+  Config:
+    Name: {^Parameter: {Name: "Name", Default: "Aggregation"}}
+    Type: RunCommand
```

Comment:
Do you care about exhausting the cursor here? If so, then the collection scanner actor will work.

Reply:
I'd prefer not to change the workload since we know it repros SERVER-29446 in the current state. I think this could be worth investigating after SERVER-29446 is fixed.